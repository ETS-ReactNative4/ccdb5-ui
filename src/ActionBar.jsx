import React from 'react';
import { connect } from 'react-redux'
import { FormattedNumber } from 'react-intl';
import './ActionBar.less';

export class ActionBar extends React.Component {
  render() {
    return (
        <summary className="action-bar">
          <div>{ this.props.hits === this.props.total
            ? (<h2>
                 Showing&nbsp;
                 <FormattedNumber value={this.props.total} />
                 &nbsp;total complaints
              </h2>)
            : (<h2>
                Showing&nbsp;
                <FormattedNumber value={this.props.hits} />
                &nbsp;matches out of&nbsp;
                <FormattedNumber value={this.props.total} />
                &nbsp;total complaints</h2>)
          }
          </div>
          <div>Page | Sort | Export</div>
        </summary>
    );
  }
}

export const mapStateToProps = state => {
  return {
    hits: state.results.total,
    total: state.results.doc_count
  }
}

export const mapDispatchToProps = dispatch => {
  return {
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ActionBar)