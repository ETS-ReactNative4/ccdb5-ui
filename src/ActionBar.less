@import (less) "base.less";

.action-bar {
    padding: @gutter-normal;
    border: 1px solid @gray-40;
    border-left: 0;  // Don't double the border with the filters

    background-color: @green-20;
}