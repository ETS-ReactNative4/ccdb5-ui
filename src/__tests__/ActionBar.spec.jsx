import React from 'react';
import { IntlProvider } from 'react-intl';
import { ActionBar} from '../ActionBar';
import renderer from 'react-test-renderer';

describe('initial state', () => {
  it('renders without crashing', () => {
    const target = renderer.create(
      <IntlProvider locale="en">
        <ActionBar total="100" hits="10" />
      </IntlProvider>
    );

    let tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

