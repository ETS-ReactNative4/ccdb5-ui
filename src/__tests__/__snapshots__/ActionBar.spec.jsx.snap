exports[`initial state renders without crashing 1`] = `
<summary
  className="action-bar">
  <div>
    <h2>
      Showing 
      <span>
        10
      </span>
       matches out of 
      <span>
        100
      </span>
       total complaints
    </h2>
  </div>
  <div>
    Page | Sort | Export
  </div>
</summary>
`;
